lib/rules/prefer-query-by-disappearance.ts

```diff
@@ -0,0 +1,184 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+
+import { createTestingLibraryRule } from '../create-testing-library-rule';
+import {
+  getPropertyIdentifierNode,
+  isArrowFunctionExpression,
+  isCallExpression,
+  isMemberExpression,
+  isFunctionExpression,
+  isExpressionStatement,
+  isReturnStatement,
+  isBlockStatement,
+} from '../node-utils';
+
+export const RULE_NAME = 'prefer-query-wait-disappearance';
+export type MessageIds = 'preferQueryByDisappearance';
+
+export default createTestingLibraryRule<[], MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'Suggest using `queryBy*` queries when waiting for disappearance',
+      category: 'Possible Errors',
+      recommendedConfig: {
+        dom: false,
+        angular: false,
+        react: false,
+        vue: false,
+      },
+    },
+    messages: {
+      preferQueryByDisappearance:
+        'Prefer using queryBy* when waiting for disappearance',
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+
+  create(context, _, helpers) {
+    function isWaitForElementToBeRemoved(node: TSESTree.CallExpression) {
+      const identifierNode = getPropertyIdentifierNode(node);
+
+      if (!identifierNode) {
+        return false;
+      }
+
+      return helpers.isAsyncUtil(identifierNode, ['waitForElementToBeRemoved']);
+    }
+
+    function isReportableExpression(node: TSESTree.LeftHandSideExpression) {
+      let argumentProperty;
+
+      if (isMemberExpression(node)) {
+        argumentProperty = getPropertyIdentifierNode(node.property);
+      } else {
+        argumentProperty = getPropertyIdentifierNode(node);
+      }
```

Comment:
This can be simplified

```suggestion
      const argumentProperty = isMemberExpression(node)
        ? getPropertyIdentifierNode(node.property)
        : getPropertyIdentifierNode(node);
```